### ISSUE TITLE
lsp-metals treeview hogs the cursor when refreshing

### ISSUE THREAD

**Issue by julianmichael** [emacs-lsp/lsp-metals#16]
**Describe the bug**
While code is compiling and the lsp-metals treeview is refreshing with compile progress for the modules in a project, it messes with the cursor: Whether I am navigating another buffer or trying to type in a command, on each refresh of the treeview, the cursor is stolen away into the treeview buffer. This prevents me from doing basically anything while code is compiling.

**To Reproduce**
Open a scala project which takes a non-trivial amount of time to compile. Edit and save a file to trigger a compile. Then try to navigate the file's buffer, execute a search within file, etc.—the cursor gets messed up on each refresh.

In case it matters, I'm using Spacemacs (develop) in Evil mode. So that might affect something related to the cursor movement. Idk.

**Expected behavior**
I expect the treeview buffer to refresh itself silently, without interfering with cursor movement or commands.

I'm not sure what a fix would look like because I don't understand what is causing the cursor to get claimed by the lsp-metals treeview buffer.

**Comment by kurnevsky** [comment]
Just to clarify - it happens when your main file buffer is focused and you are editing it, right? I didn't see such problem...

**Comment by dsyzling** [comment]
Sorry distracted with work at the moment. Since the internals of treemacs has changed and is causing issues with the metals-treeview implementation, I'm going to have try and recreate these issues once I've re-implemented on the new treemacs api. I'll also reassess some of the implementation decisions.

**Comment by julianmichael** [comment]
@kurnevsky yes, that's right. As in #14, I've worked around it for now by just not using the treemacs view. Since @dsyzling is already planning changes there, we can just wait and I can update and report back after those.

For the record, I'm on emacs 26.3, Spacemacs develop ([`c6634203`](https://github.com/syl20bnr/spacemacs/tree/c66342039947f1c44c7d1a38054e57998361224d)), and my .spacemacs file is [here](https://github.com/julianmichael/dotfiles/blob/72c1e4ff46d1814686f4a9a56c803d61ef20e040/files/spacemacs).
